api/db-cdr/generate-cdr/make-bq-denormalized-tables.sh

```diff
@@ -0,0 +1,427 @@
+#!/bin/bash
+
+# This generates big query denormalized tables.
+
+set -xeuo pipefail
+IFS=$'\n\t'
+
+
+# get options
+# --project=all-of-us-workbench-test *required
+
+# --cdr=cdr_version ... *optional
+USAGE="./generate-clousql-cdr/make-bq-denormalized-tables.sh --bq-project <PROJECT> --bq-dataset <DATASET>"
+USAGE="$USAGE --cdr-version=YYYYMMDD"
+
+while [ $# -gt 0 ]; do
+  echo "1 is $1"
+  case "$1" in
+    --bq-project) BQ_PROJECT=$2; shift 2;;
+    --bq-dataset) BQ_DATASET=$2; shift 2;;
+    -- ) shift; break ;;
+    * ) break ;;
+  esac
+done
+
+
+if [ -z "${BQ_PROJECT}" ]
+then
+  echo "Usage: $USAGE"
+  exit 1
+fi
+
+if [ -z "${BQ_DATASET}" ]
+then
+  echo "Usage: $USAGE"
+  exit 1
+fi
+
+# Check that bq_dataset exists and exit if not
+datasets=$(bq --project=$BQ_PROJECT ls)
+if [ -z "$datasets" ]
+then
+  echo "$BQ_PROJECT.$BQ_DATASET does not exist. Please specify a valid project and dataset."
+  exit 1
+fi
+if [[ $datasets =~ .*$BQ_DATASET.* ]]; then
+  echo "$BQ_PROJECT.$BQ_DATASET exists. Good. Carrying on."
+else
+  echo "$BQ_PROJECT.$BQ_DATASET does not exist. Please specify a valid project and dataset."
+  exit 1
+fi
+
+# Check that bq_dataset exists and exit if not
+datasets=$(bq --project=$BQ_PROJECT ls)
+if [ -z "$datasets" ]
+then
+  echo "$BQ_PROJECT.$BQ_DATASET does not exist. Please specify a valid project and dataset."
+  exit 1
+fi
+re=\\b$BQ_DATASET\\b
+if [[ $datasets =~ $re ]]; then
+  echo "$BQ_PROJECT.$BQ_DATASET exists. Good. Carrying on."
+else
+  echo "$BQ_PROJECT.$BQ_DATASET does not exist. Please specify a valid project and dataset."
+  exit 1
+fi
+
+# Create bq tables we have json schema for
+schema_path=generate-cdr/bq-schemas
+create_tables=(person_all_events person_condition person_device person_drug person_measurement person_observation person_procedure person_visit)
+for t in "${create_tables[@]}"
+do
+    bq --project=$BQ_PROJECT rm -f $BQ_DATASET.$t
```

Comment:
I see you remove the tables first. How long does this take to run ? Wonder if you want to check if they exist and skip it if they exist ?

Reply:
Maybe be an issue if we run this more than once on same cdr as we do in development now.

Reply:
Removing the tables takes seconds. I think the script should recreate the tables(takes about 20 mins).